### ISSUE TITLE
Vagrant VM not complete on install

### ISSUE THREAD

**Issue by thresherdj** [sillsdev/rapuma#7]
I'm having a problem getting the publishing system VM to install. Everything goes okay until near the end when I get this error:
==> default: Running provisioner: shell...
    default: Running: inline script
==> default: stdin: is not a tty
==> default: Cloning into '/home/vagrant/src/web-languageforge'...
==> default: fatal: unable to access 'https://github.com/sil-student-projects/web-languageforge.git/': Could not resolve host: github.com
==> default: Cloning into '/home/vagrant/src/web-scriptureforge'...
==> default: fatal: unable to access 'https://github.com/sil-student-projects/web-scriptureforge.git/': Could not resolve host: github.com
==> default: /tmp/vagrant-shell: line 14: cd: /home/vagrant/src/web-languageforge/deploy/: No such file or directory
==> default: fatal: Not a git repository (or any of the parent directories): .git
==> default: ERROR: the playbook: playbook_lxde.yml could not be found
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

The url (https://github.com/sil-student-projects/web-scriptureforge.git/) is good but the VM can't seem to access it. What's wrong?

**Comment by Steven-Harris** [comment]
@thresherdj Are you able to run `ssh clone https://github.com/sil-student-projects/web-scriptureforge.git/` from your host machine?

**Comment by thresherdj** [comment]
@Steven-Harris - Yes, I can put that in the browser and hit that page.

**Comment by Steven-Harris** [comment]
@thresherdj, not browser, but can you `ssh clone`` the project somewhere?

It appears that port 22 is blocked on your network or machine and you can not pull the source code over ssh.

**Comment by thresherdj** [comment]
Yes, I think you are right. any tips on how to open port 22?

**Comment by Steven-Harris** [comment]
No not really, it is all dependent on the firewall that you are using. Do you know what firewall you're using?

**Comment by thresherdj** [comment]
Not at this point, things have pretty much worked. I'll look into it.

**Comment by thresherdj** [comment]
I think the firewall is off. The error I get when I try to clone is:
Could not resolve hostname clone: Name or service not known
I can clone the Rapuma project, why can't I clone this one?

**Comment by thresherdj** [comment]
Although, I've never cloned through ssh, I just use git clone

**Comment by Steven-Harris** [comment]
_facepalm_ sorry i meant git clone not ssh clone

**Comment by Steven-Harris** [comment]
with http git clone port 443 will have to be open.. which if url is loading in the browser from the host, then that port is open. Maybe, the vagrant file is doing `git clone git@github.com:sil-student-projects/web-scriptureforge.git` in that case it would use port 22..

Any chance you could just take down your firewall while you try to install?

**Comment by thresherdj** [comment]
I don't think I was even running a firewall. I ran:
sudo ufw disable
then:
ssh clone https://github.com/sil-student-projects/web-scriptureforge.git/
and I got:
Cloning into 'web-scriptureforge'...
The authenticity of host 'github.com (192.30.252.128)' can't be established.
RSA key fingerprint is 16:27:ac:a5:76:28:2d:36:63:1b:56:4d:eb:df:a6:48.
Are you sure you want to continue connecting (yes/no)? y
Please type 'yes' or 'no': yes
Warning: Permanently added 'github.com,192.30.252.128' (RSA) to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by Steven-Harris** [comment]
sorry I don't think you saw my message earlier.. I didn't mean to type ssh clone..

you can use `git clone https://github.com/sil-student-projects/web-scriptureforge.git`
or `git clone git@github.com:sil-student-projects/web-scriptureforge.git`

**Comment by thresherdj** [comment]
ah, much better, the first address works good

**Comment by thresherdj** [comment]
Now I'll try it in the VM

**Comment by thresherdj** [comment]
No, doesn't work in the VM. When I run:
 git clone https://github.com/sil-student-projects/web-scriptureforge.git
I get:
Cloning into 'web-scriptureforge'...
fatal: unable to access 'https://github.com/sil-student-projects/web-scriptureforge.git/': Could not resolve host: github.com
With the second command:
git clone git@github.com:sil-student-projects/web-scriptureforge.git
I get:
Cloning into 'web-scriptureforge'...
ssh: Could not resolve hostname github.com: Name or service not known
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

No joy yet :-(

**Comment by thresherdj** [comment]
BTW, after we figure this out, take a look at issue #8

**Comment by Steven-Harris** [comment]
@cambell-prince, I run the vagrantFile for myself and it does appear to be broken. Here is the infor when I tried for the first time

> Bringing machine 'default' up with 'virtualbox' provider...
> ==> default: Box 'trusty-webdev-amd64' could not be found. Attempting to find and install...
>     default: Box Provider: virtualbox
>     default: Box Version: >= 0
> ==> default: Box file was not detected as metadata. Adding it directly...
> ==> default: Adding box 'trusty-webdev-amd64' (v0) for provider: virtualbox
>     default: Downloading: http://downloads.sil.org/vagrant/trusty-webdev-amd64.box
> ==> default: Successfully added box 'trusty-webdev-amd64' (v0) for 'virtualbox'!
> ==> default: Importing base box 'trusty-webdev-amd64'...
> ==> default: Matching MAC address for NAT networking...
> ==> default: Setting the name of the VM: Projects_default_1460218522934_27694
> ==> default: Clearing any previously set network interfaces...
> ==> default: Preparing network interfaces based on configuration...
>     default: Adapter 1: nat
> ==> default: Forwarding ports...
>     default: 22 => 2222 (adapter 1)
> ==> default: Running 'pre-boot' VM customizations...
> ==> default: Booting VM...
> ==> default: Waiting for machine to boot. This may take a few minutes...
>     default: SSH address: 127.0.0.1:2222
>     default: SSH username: vagrant
>     default: SSH auth method: private key
>     default: Warning: Connection timeout. Retrying...
>     default: 
>     default: Vagrant insecure key detected. Vagrant will automatically replace
>     default: this with a newly generated keypair for better security.
>     default: 
>     default: Inserting generated public key within guest...
>     default: Removing insecure key from the guest if it's present...
>     default: Key inserted! Disconnecting and reconnecting using new SSH key...
> ==> default: Machine booted and ready!
> ==> default: Checking for guest additions in VM...
>     default: The guest additions on this VM do not match the installed version of
>     default: VirtualBox! In most cases this is fine, but in rare cases it can
>     default: prevent things such as shared folders from working properly. If you see
>     default: shared folder errors, please make sure the guest additions within the
>     default: virtual machine match the version of VirtualBox you have installed on
>     default: your host and reload your VM.
>     default: 
>     default: Guest Additions Version: 4.3.10
>     default: VirtualBox Version: 5.0
> ==> default: Mounting shared folders...
>     default: /vagrant => /home/superman/Projects
> ==> default: Running provisioner: shell...
>     default: Running: inline script
> ==> default: stdin: is not a tty
> ==> default: Cloning into '/home/vagrant/src/web-languageforge'...

**Comment by Steven-Harris** [comment]
Dennis is did you try those clone commands on your host machine?

**Comment by Steven-Harris** [comment]
Never mind I remember now that you did

**Comment by thresherdj** [comment]
@Steven-Harris , @cambell-prince : Any word on this? I'm kind of stuck without being able to pull the web-scriptureforge project into the VM. It seems the VM, in general, has a hard time pulling data. I've been testing the new version of the Rapuma install and it seem to work predictably but without being able to try it along with scriptureforge, I'm never going to know if I have it right or not.

**Comment by Steven-Harris** [comment]
I haven't heard from @cambell-prince. At this point you might go ahead and create VM manually and install Ubuntu or mint and install scriptureforge manually because I know what works.

**Comment by thresherdj** [comment]
I already have one but I can't seem to pull in scriptureforge on that either. It starts and then stalls out. Rapuma seems to pull ok though. I'll keep trying and see if I can figure it out.

**Comment by Steven-Harris** [comment]
hmm that would lead me to suspect that it has something to do with your
host.. Just for fu, try cloning from my forked repo of it
https://github.com/Steven-Harris/web-scriptureforge.git

**Comment by thresherdj** [comment]
I tried downloading the zip inside my VM and it stalls out as well. I'm wondering if it is my ISP that is the problem. I don't think they play nice with VMs. Though that doesn't seem to be a problem with my standard install. It down loads okay from my standard install.

**Comment by thresherdj** [comment]
@Steven-Harris , @cambell-prince : Ok, I was right, it is my ISP. I am in a different location now with a completely different ISP and downloading works fine. The issue with the Vagrant VM is still the same.

**Comment by Steven-Harris** [comment]
I would abandon the vagrant install and go to installing the app manually
for the repo

**Comment by thresherdj** [comment]
Yah, I figured that. Please remind me where the install instructions are.

**Comment by Steven-Harris** [comment]
Just clone the repo on a mint or Ubuntu system/vm and the install
instructions are in the README

**Comment by thresherdj** [comment]
Ah, thanks. I'll work on it.

**Comment by thresherdj** [comment]
@Steven-Harris : When I try to deploy (line 35) on my Mint system with:
ansible-playbook -i hosts playbook_mint.yml --limit localhost -K

I get this:
ERROR: cannot find role in /home/dennis/src/xForge/web-languageforge/deploy/roles/fix or /home/dennis/src/xForge/web-languageforge/deploy/fix or /etc/ansible/roles/fix

What went wrong?

**Comment by Steven-Harris** [comment]
You need to clone using 'git clone --recurse-submodules. I would start from
the top of the README and follow it exactly

**Comment by thresherdj** [comment]
That's what I thought I was doing and when I got to the command on line 35 I got that error. I'll try it again.

**Comment by Steven-Harris** [comment]
you might have already done it but you could try `git pull --recurse-submodules` in the root of the repo

**Comment by thresherdj** [comment]
On line 19, it says this:
"If you want to run an independant repo for scriptureforge, clone its repo also..."

Do I want that?

**Comment by Steven-Harris** [comment]
Yeah give that a shot

**Comment by thresherdj** [comment]
Stymied again! I run ran lines 34 and 35 and got:
ERROR: cannot find role in /home/dennis/src/xForge/web-languageforge/deploy/roles/fix or /home/dennis/src/xForge/web-languageforge/deploy/fix or /etc/ansible/roles/fix

**Comment by thresherdj** [comment]
I also ran: git pull --recurse-submodules

**Comment by Steven-Harris** [comment]
if you install teamview on the machine I could remote in and help you out just send me the cerd @ stevenharrisdev@gmail.com

**Comment by thresherdj** [comment]
Teamview on a VM, that sound's interesting. I'll give it a try.

**Comment by thresherdj** [comment]
ID: 124 397 523
PW: 3177

**Comment by thresherdj** [comment]
The chat on teamviewer in the VM doesn't work for me. The keyboard input seems to go to the terminal, not the chat window.

**Comment by thresherdj** [comment]
I can read your messages fine, just can't respond with teamviewer chat.

**Comment by Steven-Harris** [comment]
it kicked me from the session

**Comment by thresherdj** [comment]
Sorry, my fault. trying to get the silly window out of the way.

**Comment by thresherdj** [comment]
Can you log back on?

**Comment by Steven-Harris** [comment]
Well I am not sure why it is giving us the error... but for your testing you can try creating a project and then see if there is a rapuma project at /var/lib/rapuma/Publishing ... If you are able to do that then rapuma is working correctly with our project. 

The code looks a bit different than mine so I will make sure that my changes are pushed. You might pull again tomorrow to get my new stuff

let me know if you have any questions

**Comment by thresherdj** [comment]
Ok, sounds like a plan.

**Comment by Steven-Harris** [comment]
it's ready to pull

**Comment by megahirt** [comment]
@irahopkinson can you comment on Dennis's issue, specifically (from above):

When I try to deploy (line 35) on my Mint system with:
ansible-playbook -i hosts playbook_mint.yml --limit localhost -K

I get this:
ERROR: cannot find role in /home/dennis/src/xForge/web-languageforge/deploy/roles/fix or /home/dennis/src/xForge/web-languageforge/deploy/fix or /etc/ansible/roles/fix

What went wrong?

**Comment by Steven-Harris** [comment]
@megahirt I should note that I was about to get Dennis up and running when i teamviewed into his vm

**Comment by irahopkinson** [comment]
@thresherdj what's your ansible version `ansible --version`
Do you have the folder `/home/dennis/src/xForge/web-languageforge/deploy/roles_common/fix`?
Ah, I think your problem is your `ansible.cfg`. The `README_DEVELOPERS.md` should put you straight. In case your branch doesn't have the latest updates to this file, look here https://github.com/sillsdev/web-languageforge/blob/master/README_DEVELOPERS.md#ansible-setup and do the last line from this section **Ansible Setup**. I.e. "...merge the contents of `deploy/default_ansible.cfg` into `/etc/ansible/ansible.cfg`."

**Comment by thresherdj** [comment]
@irahopkinson : Hey there!
ansible version 1.5.4
yes, I have the fix folder
I'll try merging the two .cfg files
thanks!

**Comment by thresherdj** [comment]
I'm going to try adding the default_ansible.cfg to the .ansible folder to see if the settings there overwrite the ones in etc. This may end poorly.

**Comment by thresherdj** [comment]
Well, at least no small explosions or smoke was involved. ;-)

I'll try merging them but lunch is about over now so it will have to wait until later.

**Comment by Steven-Harris** [comment]
@thresherdj, I found the solution to the bug we were getting on the web app. 

```
cd ~/src/web-scriptureforge/src
mkdir assets
chown -R dennis:www-data assets
chmod -R 775 assets
```

**Comment by irahopkinson** [comment]
@thresherdj Keep up the good work!
Ansible v1.5.4 is likely to have issues as it is too old. However, the latest v2 is broken for our purposes (last time I checked) so use v1.9.4. Check here https://github.com/sillsdev/web-languageforge/blob/master/README_DEVELOPERS.md#ansible-setup for how to update it. I.e.

```
sudo apt-get install python-pip
sudo pip install ansible==1.9.4
```

Then do the merge into the `ansible.cfg` file.
